internal/app/service.go

```diff
@@ -119,10 +118,16 @@ func (svc *Service) AddRoute(route string, handler func(nethttp.ResponseWriter,
 
 // AddBackgroundPublisher will create a channel of provided capacity to be
 // consumed by the MessageBus output and return a publisher that writes to it
-func (svc *Service) AddBackgroundPublisher(capacity int) interfaces.BackgroundPublisher {
-	bgchan, pub := newBackgroundPublisher(capacity)
+func (svc *Service) AddBackgroundPublisher(capacity int) (interfaces.BackgroundPublisher, error) {
```

Comment:
Do we want another version that allows for the topic to be specified rather than using the configure MessageBus PublishTopic?